site/docs/patterns/index.mdx

```diff
@@ -0,0 +1,524 @@
+
+<ImageSwitcher
+  images={[
+    { src: "/img/patterns/forms/form_layout_1.png", alt: "" },
+    { src: "/img/patterns/forms/form_layout_2.png", alt: "" },
+  ]}
+  label="Show spacing"
+/>
+
+\*You can find more information on the Salt spacing system here.
+
+<ExampleContainer type="positive">
+- Align the width of a Form Field to the number of columns that best reflect the length of its expected text.
+- Use the Figma "Salt Layout Grid" style library, which aligns to Salt Breakpoints. Simply pick the style that fits to your density and frame width.
+- Use "Compact layout" for very data-dense forms.
+</ExampleContainer>
+
+## Form container
+
+<Diagrams>
+  <Diagram
+    src="/img/patterns/forms/form_container_1.png"
+    alt="Example showing full page form container"
+    contentPosition="top"
+  >
+    ### Full page
+    Embed your form in the application layout when the form is:
+    - Long and complex, or requires multiple columns.
+    - A self-container workflow, such as a log-in page—which may include an element of background branding.
+  </Diagram>
+  <Diagram
+    src="/img/patterns/forms/form_container_2.png"
+    alt="Example showing overlay form container"
+    contentPosition="top"
+  >
+    ### Overlay (Dialog/Drawer)
+    Typically displays smaller forms that:
+    - Collect information to inform a larger workflow.
+    - Are displayed on application boot (i.e., as the app loads).
+    - Enable access/feedback (i.e., credentials input, reviews).
+  </Diagram>
+</Diagrams>
+
+<ExampleContainer type="positive">
+  In both cases, align the form to a layout grid within the viewport.
+</ExampleContainer>
+
+## Field vs. background fill
+
+Salt provides primary and secondary variants of both Form Field and backgrounds.
+
+To keep busy forms as simple as possible, choose a combination of these fills to achieve visual hierarchy. Rather than adding additional background colours, if further visual separation is required, use:
+
+1. "Headings"
+2. "Horizontal separators"
+3. "Progressive disclosure"
+4. "Columns"
+
+Choose the combination that best fits:
+
+- The visual priority of the form (i.e., is it the primary focus?).
+- The volume, and type of data, being collected.
+- The user familiarity with the form.
+
+<Diagrams>
+  <Diagram
+    src="/img/patterns/forms/field_background_1.png"
+    alt="Example showing primary field on primary background"
+    contentPosition="top"
+    border
+  >
+    ### 1a Primary field on primary background
+    Effective for a form that:
+    - Is 'non-specialist' or infrequently used.
+    - Isn't space constrained/doesn't need to be entirely visible.
+    - Has minimal sections/columns.
+  </Diagram>
+  <Diagram
+    src="/img/patterns/forms/field_background_2.png"
+    alt="Example showing secondary field on secondary background"
+    contentPosition="top"
+  >
+    ### 1b Secondary field on secondary background
```

Comment:
Shouldn't this be primary field on secondary background?

Reply:
Updated thanks